activerecord/lib/active_record/connection_adapters/postgresql_adapter.rb

```diff
@@ -608,6 +608,10 @@ def table_name_length
         max_identifier_length - "_pkey".length
       end
 
+      def table_name_too_long_error_message(table_name)
+        "Table name '#{table_name}' is too long; PostgreSQL allows a maximum length of #{max_identifier_length} characters, but we also need to allow for the `_pkey` suffix that PostgreSQL adds to table names when creating default indexes, so our effective limit is #{table_name_length} characters"
```

Comment:
While this message is _accurate_, it feels like a rather unergonomic info-dump instead of a short actionable error message.

While I can appreciate the confusion that our enforced limit differs from PostgreSQL's, I still don't feel like this is the place to tell that story.

Reply:
@matthewd agreed, it's a brain dump of an error message.  This feels like a short term fix to me. I don't think we should be limiting table names to 58 characters given it's not actually a Postgres limitation. The issue is actually with index names. A table name of 63 characters will work fine so I think we should allow it and instead fix the index names if they are too long.

I didn't want to pollute this PR and was going to submit a separate issue to allow table names of 63 characters again.

I thought improving the error message might be something we could do quickly to help people upgrading to Rails 7.1 as it caused me a bit of confusion.

Depending on what everyone else thinks, I'm happy to submit a separate PR to allow table names up to 63 characters again.